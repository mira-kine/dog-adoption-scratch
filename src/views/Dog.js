import React from 'react';
import { useState, useEffect } from 'react';
import { useHistory, useParams } from 'react-router-dom';
import { deleteDog, fetchDogById } from '../services/dogs';
import DogDetail from '../components/DogDetail';
import './Dog-View.css';
import Header from '../components/Header/Header';

export default function Dog() {
  const [dog, setDog] = useState([]);
  const [loading, setLoading] = useState(true);
  const params = useParams();
  const history = useHistory();

  useEffect(() => {
    let timer;
    const fetchData = async () => {
      const data = await fetchDogById(params.id);
      setDog(data);
      timer = setTimeout(() => {
        setLoading(false);
      }, 100);
    };
    if (loading) {
      fetchData();
    }
    return () => {
      clearInterval(timer);
    };
  }, [params.id, loading]);

  const handleDelete = async (e) => {
    e.preventDefault();
    await deleteDog(dog.id);
    alert("You've successfully deleted your dog");
    history.push('/');
  };

  return (
    <>
      {loading && <div className="loader"></div>}
      {!loading && (
        <div>
          <DogDetail dog={dog} handleDelete={handleDelete} />
          <Header />
        </div>
      )}
    </>
  );
}
