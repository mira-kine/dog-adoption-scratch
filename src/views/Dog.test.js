import { MemoryRouter, Route } from 'react-router-dom/cjs/react-router-dom.min';
import Dog from '../views/Dog';
import { render, screen } from '@testing-library/react';

test('should render an individual dog', async () => {
  // render the component
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/6']}>
      <Route path="/dogs/:id">
        <Dog />
      </Route>
    </MemoryRouter>
  );
  await screen.findByText('Mookie');
  expect(container).toMatchSnapshot();
});
