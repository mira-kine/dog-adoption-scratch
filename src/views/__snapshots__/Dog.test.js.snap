// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an individual dog 1`] = `
<div>
  <div>
    <div
      class="dog-card"
    >
      <h1>
        Pup
      </h1>
      <img
        src="https://placedog.net/500?id=30"
      />
      <p>
        Pup
         is a(n) 
        3
         year old 
      </p>
      <p />
      <button>
        <a
          href="/dogs/9/edit"
        >
          Edit
        </a>
      </button>
      <button>
        Delete
      </button>
    </div>
    <div>
      <a
        href="/"
      >
        Home
      </a>
    </div>
  </div>
</div>
`;
